python-avd/pyavd/_cv/client/async_decorators.py

```diff
@@ -150,3 +102,205 @@ async def wrapper_cv_version(*args: Any, **kwargs: Any) -> list:
+    """
+    Decorator used to handle execution of the async gRPC calls towards CloudVision.
+
+    Retries an async method upon getting gRPC Status.UNAVAILABLE (14) using exponential backoff mechanism and max retry limit.
+    Converts GRPCError or AsyncioTimeoutError instances to an instance of the relevant subclass of CVClientException.
+    Splits gRPC messages into smaller chunks (based on reported maximum supported size) if Status.RESOURCE_EXHAUSTED is received.
+
+    Args:
+        max_retries (int): Maximum number of retry attempts for Status.UNAVAILABLE. Total attempts = 1 + max_retries.
+        initial_delay (int): Initial delay in seconds before the first retry.
+        factor (int): Multiplier for the delay in subsequent retries.
+        list_field (str): Name of the parameter to be split if Status.RESOURCE_EXHAUSTED is received.
+        min_items_for_splitting_attempt (int): Minimum length of the item that we'll still try to split.
+    """
+
+    def __init__(
+        self,
+        max_retries: int = 5,
+        initial_delay: int = 1,
+        factor: int = 2,
+        list_field: str | None = None,
+        min_items_for_splitting_attempt: int = 2,
+    ) -> None:
+        self.max_retries: int = max_retries
+        self.initial_delay: int = initial_delay
+        self.factor: int = factor
+        self.list_field: str | None = list_field
+        self.min_items_for_splitting_attempt = max(2, min_items_for_splitting_attempt)
+        self.func: Callable
+        self.func_signature: Signature
+
+    def __call__(self, func: Callable) -> Callable:
+        self.func = func
+        self.func_signature = signature(func)
+
+        if self.list_field:
+            return_annotation = (
+                list
+                if type(self.func_signature.return_annotation) is str and self.func_signature.return_annotation.startswith("list")
+                else self.func_signature.return_annotation
+            )
+            if return_annotation is not list and get_origin(return_annotation) is not list:
+                msg = (
+                    f"GRPCRequestHandler decorator is unable to bind to the function '{func.__name__}'. "
+                    f"Expected a return type of 'list'. Got '{return_annotation}'."
+                )
+                raise TypeError(msg)
+
+        @wraps(func)
+        async def wrapper(*args: Any, **kwargs: Any) -> Callable:
+            return await self._execute_with_splitting(args, kwargs)
+
+        return wrapper
+
+    async def _execute_single_call_with_retries(self, call_args: tuple, call_kwargs: dict) -> Any:
+        """Executes a single call to self.func with retry logic for gRPC UNAVAILABLE."""
+        func_name = self.func.__name__
+
+        for attempt in range(1, self.max_retries + 2):
+            try:
+                return await self.func(*call_args, **call_kwargs)
+            except Exception as e:  # noqa: PERF203
+                if isinstance(e, GRPCError):
```

Comment:
let's negate this check so we don't have to scroll many lines to figure out the other outcome.